Automattic/wp-calypso
# Domains: Delete domain-only site when the corresponding domain is canceled

@aidvu created a patch to handle this on the API. This PR updates `RemovePurchase` such that a domain removed from a domain-only site removes the site from `sites-list` and is no longer set as the selected site.

**Testing**
- ~~Apply D4334-code and~~ sandbox the API.
- Purchase a domain from `/start/domain-first`.
- Visit `/me/purchases` and cancel the domain.
- Visit `/me/purchases` again and remove the domain from your account.
- Assert that you are redirected back to `/me/purchases` once the domain is removed and that `My Sites` no longer links to `/domains/manage/:site_slug_for_the_site_you_just_deleted`.
